File: Assets/Import_out/Uzil/API/ProfileSave.cs
using Uzil;

using UzProfileSave = Uzil.UserData.ProfileSave;


namespace UZAPI {

public class ProfileSave {

	/*======================================Constructor==========================================*/

	/*=====================================Static Members========================================*/

	/*=====================================Static Funciton=======================================*/

	/*=========================================Members===========================================*/

	/*========================================Components=========================================*/

	/*==========================================Event============================================*/

	/*======================================Unity Function=======================================*/

	/*========================================Interface==========================================*/

	/** 是否存在 */
	public static bool IsExist (string path, string key) {
		return UzProfileSave.main.IsExist(path, key);
	}

	/** 設置 字串 */
	public static bool SetStr (string path, string key, string val) {
		return UzProfileSave.main.SetStr(path, key, val);
	}

	/** 設置 數字 */
	public static bool SetNum (string path, string key, float val) {
		return UzProfileSave.main.SetNum(path, key, val);
	}

	/** 設置 布林 */
	public static bool SetBool (string path, string key, bool val) {
		return UzProfileSave.main.SetBool(path, key, val);
	}

	/** 設置 物件 */
	public static bool SetObj (string path, string key, string val_json) {
		return UzProfileSave.main.SetObj(path, key, DictSO.Json(val_json));
	}

	/** 設置 物件 */
	public static bool SetList (string path, string key, string val_json) {
		return UzProfileSave.main.SetList(path, key, DictSO.List<object>(val_json));
	}
	
	/** 取得 字串 */
	public static string GetStr (string path, string key) {
		return UzProfileSave.main.GetStr(path, key);
	}

	/** 取得 數字 */
	public static int? GetInt (string path, string key) {
		return UzProfileSave.main.GetInt(path, key);
	}

	/** 取得 數字 */
	public static double? GetFloat (string path, string key) {
		return UzProfileSave.main.GetFloat(path, key);
	}

	/** 取得 布林 */
	public static bool? GetBool (string path, string key) {
		return UzProfileSave.main.GetBool(path, key);
	}

	/** 移除 */
	public static bool Remove (string path, string key) {
		return UzProfileSave.main.Remove(path, key);
	}

	/** 刪除 */
	public static bool Delete (string path) {
		return UzProfileSave.main.Delete(path);
	}

	/*=====================================Public Function=======================================*/

	/*===================================Protected Function======================================*/

	/*====================================Private Function=======================================*/

}

}

File: AppData/Mods/BUILT-IN/_lua/LuaScript/Uzil/ProfileSave.lua
local ProfileSave = {};

function ProfileSave.inst (path)

	local this = {};
	this.path = path;

	this.isExist = function (key)
		return ProfileSave.isExist(this.path, key);
	end

	this.setStr = function (key, val)
		return ProfileSave.setStr(this.path, key, val);
	end

	this.setNum = function (key, val)
		return ProfileSave.setNum(this.path, key, val);
	end

	this.setBool = function (key, val)
		return ProfileSave.setBool(this.path, key, val);
	end

	this.setTable = function (key, val)
		return ProfileSave.setTable(this.path, key, val);
	end

	this.getStr = function (key)
		return ProfileSave.getStr(this.path, key);
	end

	this.getInt = function (key)
		return ProfileSave.getInt(this.path, key);
	end

	this.getFloat = function (key)
		return ProfileSave.getFloat(this.path, key);
	end

	this.getBool = function (key)
		return ProfileSave.getBool(this.path, key);
	end

	this.getTable = function (key)
		return ProfileSave.getTable(this.path, key);
	end

	this.remove = function (key)
		return ProfileSave.remove(this.path, key);
	end

	this.delete = function ()
		return ProfileSave.delete(this.path);
	end

	return this;
end
ProfileSave.file = ProfileSave.inst;

--[[ 是否 存在 ]] -- bool
function ProfileSave.isExist (path, key)
   return tobool(UZAPI.ProfileSave.IsExist(path, key));
end

--[[ 設置 字串 ]] -- bool
function ProfileSave.setStr (path, key, val)
	return tobool(UZAPI.ProfileSave.SetStr(path, key, val));
end

--[[ 設置 數字 ]] -- bool
function ProfileSave.setNum (path, key, val)
	return tobool(UZAPI.ProfileSave.SetNum(path, key, val));
end

--[[ 設置 布林 ]] -- bool
function ProfileSave.setBool (path, key, val)
	return tobool(UZAPI.ProfileSave.SetBool(path, key, val));
end

--[[ 設置 物件 ]] -- bool
function ProfileSave.setTable (path, key, val)
	if (Util.isArray(val)) then
		return tobool(UZAPI.ProfileSave.SetList(path, key, Json.encode(val)));
	else
		return tobool(UZAPI.ProfileSave.SetObj(path, key, Json.encode(val)));
	end
end

--[[ 取得 字串 ]] -- string
function ProfileSave.getStr (path, key)
	return UZAPI.ProfileSave.GetStr(path, key);
end

--[[ 取得 變數 ]] -- int
function ProfileSave.getInt (path, key)
	return UZAPI.ProfileSave.GetInt(path, key);
end

--[[ 取得 變數 ]] -- float
function ProfileSave.getFloat (path, key)
	return UZAPI.ProfileSave.GetFloat(path, key);
end


--[[ 取得 變數 ]] -- bool
function ProfileSave.getBool (path, key)
	return UZAPI.ProfileSave.GetBool(path, key);
end

--[[ 取得 物件 ]] -- table
function ProfileSave.getTable (path, key)
	local obj_str = UZAPI.ProfileSave.GetStr(path, key);
	local obj = Json.decode(obj_str);
	return obj;
end

--[[ 移除 ]] -- bool
function ProfileSave.remove (path, key)
	return UZAPI.ProfileSave.Remove(path, key);
end

--[[ 刪除 ]] -- bool
function ProfileSave.delete (path)
	return UZAPI.ProfileSave.Delete(path);
end


return ProfileSave;